composer.json

```diff
@@ -7,15 +7,20 @@
     "license": "Apache-2.0",
     "require": {
         "php": ">=5.4",
-        "google/auth": "dev-master"
+        "google/auth": "dev-master",
+        "monolog/monolog": "^1.17",
+        "phpseclib/phpseclib": "~2.0"
     },
     "require-dev": {
         "phpunit/phpunit": "3.7.*",
         "squizlabs/php_codesniffer": "~2.3"
     },
     "autoload": {
+        "psr-0": {
+            "Google_": "src/"
+        },
         "classmap": [
-            "src/"
+            "src/Google/Service/"
         ]
```

Comment:
@mtdowling upgrading to v6 will take some consideration, as the entire underlying [google auth library](https://github.com/google/google-auth-library-php) is built on `GuzzleHttp\SubscriberInterface`.

I suppose I could convert the [subscriber classes](https://github.com/google/google-auth-library-php/blob/master/src/AuthTokenFetcher.php) to implement `_invoke()`, and then create the `HandlerStack` from that, but currently the idea is the `GuzzleHttp\Client` is passed in and "authenticated", but since `'handler'` is a constructor-only argument, this won't be possible.

The user must create the `GuzzleHttp\Client` and explicitly add our auth handlers, *or* we create a default client for them. So `$googleClient->getAuthHandler()` could return the callable for them to push on their stack. This is more work for the users than I would like to get a custom Http Client, but less magical than an `authorize` method. Is this the only way?